test/distributed/_tensor/test_optimizers.py

```diff
@@ -84,23 +84,26 @@ def _assert_optimizer(
                 # Default 'rtol' and 'atol' for attr:`~torch.float32` are ``1.3e-6`` and ``1e-5``
                 self.assertEqual(p1, p2, atol=atol, rtol=rtol)
 
+    def test_optimizer_foreach_supported_types_include_DTensor(self):
+        from torch.optim.optimizer import _foreach_supported_types
+
+        self.assertTrue(DTensor in _foreach_supported_types)
+
     @with_comms
     def test_adam_1d_sharding(self):
         mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
 
         # TODO: add fused_adam support
         adam_configs = [
-            {"lr": 0.1},
+            {"lr": 0.1, "foreach": False},
```

Comment:
wondering why we turn `foreach` to False for all tests?
iiuc even if we put DTensor to `_foreach_supported_types`, if we pass `foreach=False` manually to optimizer, it would disable the foreach optimizer path too

Reply:
I am just reverting the config based on whether we have "foreach": True originally.

If we have `"foreach": True` in the config, then we remove it, as it is turned on by default. 
For the config that doesn't have `"foreach": True`, I am turning it to `"foreach": False`.
So we still have some tests for both foreach and the for-loop implementation if it makes sense.

Reply:
ohh i see, make sense